src/engine-actions/translate-swap.js

```diff
@@ -1,227 +1,80 @@
+ *                                         the Taker node we are paying
+ * @param {string} swapHash              - swap hash that will be associated
+ *                                         with the swap
+ * @param {string} amount                - Int64 string of the amount of
+ *                                         outbound currency in integer units
+ * @param {Date} maxTime                 - Absolute time representing the max
+ *                                         time for a payment to resolve
+ *                                         considering the time locks on the
+ *                                         route.
+ * @param {string} finalCltvDeltaSecs    - Int64 string of final hop time lock
+ *                                         in seconds
+ * @returns {string} Base64 string of the preimage for the swapHash
+ * @throws {PermanentSwapError} If an error is encountered while paying that we
+ * are sure did not result in a still-active outbound HTLC for the swap hash.
  */
-async function translateSwap (takerAddress, swapHash, amount, extendedTimeLockDelta) {
-  this.logger.info(`Translating swap for ${swapHash} by sending to ${takerAddress}`, { amount, extendedTimeLockDelta })
+async function translateSwap (takerAddress, swapHash, amount, maxTime, finalCltvDeltaSecs = DEFAULT_MIN_FINAL_DELTA) {
+  this.logger.info(`Translating swap for ${swapHash} by sending to ${takerAddress}`,
+    { amount, maxTime })
 
   const { publicKey } = networkAddressFormatter.parse(takerAddress)
 
-  const secondsPerBlock = this.secondsPerBlock
-
-  // We specifically use Math.ceil here to ensure that the swap succeeds by providing
-  // the terminating node (Taker) additional timelock. This allows enough time so that
-  // the Taker will not reject the HTLC as having too little timelock. We add a Block buffer
-  // to account for blocks that are mined during swap execution.
-  const finalCltvDelta = Math.ceil((DEFAULT_MIN_FINAL_DELTA + BLOCK_BUFFER) / secondsPerBlock)
-
-  const queryRoutesReq = {
-    pubKey: publicKey,
+  // Convert the maximum time into a time from now so we can convert it into
+  // blocks of relative time. We round down to a whole number of seconds to
+  // be conservative.
+  const timeFromNow = Math.floor((maxTime - new Date()) / 1000)
+
+  // Translate seconds into blocks, rounding down to ensure we can stay atomic.
+  // The LND api expects this to be an int32, which in javascript is a number.
+  const cltvLimit = parseInt(Big(timeFromNow).div(this.secondsPerBlock).round(3).toFixed(0), 10)
+
+  // Translate seconds into blocks, rounding up so that we're sure the end
+  // node will accept the payment. The LND api expects this to be an int32,
+  // so we convert to a javascript number.
+  const finalCltvDelta = parseInt(Big(finalCltvDeltaSecs).div(this.secondsPerBlock).round(0).toFixed(0), 10)
+
+  const {
+    paymentPreimage,
+    paymentError
+  } = await sendPayment({
```

Comment:
Since `sendPayment` uses `SendPaymentSync` on lnd, our original conversation about calling `TrackPayment` first followed by `sendPayment` should be correct.

Alternatively, we could change `sendPayment` from calling `SendPaymentSync` on `rpc.proto` to calling `SendPayment` on `router.proto`, which errors if the payment is already in flight or has already been paid (https://github.com/lightningnetwork/lnd/blob/f03533c67a2b2dc62c81b7c7972316d5a98e32eb/lnrpc/routerrpc/router_server.go#L206-L207).

I don't think we gain much by changing to `SendPayment` on `router.proto`, however I'm open to making those changes when we add `TrackPayment`.